src/compute/src/arrangement/manager.rs

```diff
@@ -97,20 +105,118 @@ impl TraceManager {
+/// arrangements.
+#[derive(Clone)]
+pub enum SpecializedTraceHandle {
+    Bytes9Row(
+        Vec<ColumnType>,
+        TraceRowHandle<Bytes9, Row, Timestamp, Diff>,
+    ),
+    RowRow(TraceRowHandle<Row, Row, Timestamp, Diff>),
+}
+
+impl SpecializedTraceHandle {
+    /// Obtains the logical compaction frontier for the underlying trace handle.
+    fn get_logical_compaction(&mut self) -> AntichainRef<Timestamp> {
+        match self {
+            SpecializedTraceHandle::Bytes9Row(_, handle) => handle.get_logical_compaction(),
+            SpecializedTraceHandle::RowRow(handle) => handle.get_logical_compaction(),
+        }
+    }
+
+    /// Advances the logical compaction frontier for the underlying trace handle.
+    pub fn set_logical_compaction(&mut self, frontier: AntichainRef<Timestamp>) {
+        match self {
+            SpecializedTraceHandle::Bytes9Row(_, handle) => handle.set_logical_compaction(frontier),
+            SpecializedTraceHandle::RowRow(handle) => handle.set_logical_compaction(frontier),
+        }
+    }
+
+    /// Advances the physical compaction frontier for the underlying trace handle.
+    pub fn set_physical_compaction(&mut self, frontier: AntichainRef<Timestamp>) {
+        match self {
+            SpecializedTraceHandle::Bytes9Row(_, handle) => {
+                handle.set_physical_compaction(frontier)
+            }
+            SpecializedTraceHandle::RowRow(handle) => handle.set_physical_compaction(frontier),
+        }
+    }
+
+    /// Reads the upper frontier of the underlying trace handle.
+    pub fn read_upper(&mut self, target: &mut Antichain<Timestamp>) {
+        match self {
+            SpecializedTraceHandle::Bytes9Row(_, handle) => handle.read_upper(target),
+            SpecializedTraceHandle::RowRow(handle) => handle.read_upper(target),
+        }
+    }
+
+    /// Maps the underlying trace handle to a `SpecializedArrangementImport`,
+    /// while readjusting times by `since` and `until`.
+    pub fn import_frontier_logged<'g, G, T>(
+        &mut self,
+        scope: &Child<'g, G, T>,
+        name: &str,
+        since: Antichain<Timestamp>,
+        until: Antichain<Timestamp>,
+        logger: Option<Logger>,
+        idx_id: GlobalId,
+        export_ids: Vec<GlobalId>,
+    ) -> (
+        SpecializedArrangementImport<Child<'g, G, T>, Timestamp>,
+        ShutdownButton<CapabilitySet<Timestamp>>,
+    )
+    where
+        G: Scope<Timestamp = Timestamp>,
+        T: Lattice + Refines<G::Timestamp>,
+    {
+        match self {
+            SpecializedTraceHandle::Bytes9Row(key_types, handle) => {
+                let (oks, oks_button) =
+                    handle.import_frontier_core(&scope.parent, name, since, until);
+                let oks = if let Some(logger) = logger {
+                    oks.log_import_frontiers(logger, idx_id, export_ids)
+                } else {
+                    oks
```

Comment:
uncovered

Reply:
For the specific example above, we'd need to create a replica with gathering of introspection data turned off in the new test `specialized_arrangements.td`. But this illustrates the more general problem with this PR: We already have (or should have!) other tests that cover this case. But we cannot make use of them until we get further along enough to turn on the feature flag in CI.